.github/workflows/builder_source-archive_slsa3.yml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA Source archive builder
+
+permissions: {}
+
+on:
+  workflow_call:
+    inputs:
+      formats:
+        description: >
+          A list of archive formats to generate. Supported: "zip" and "tar.gz".
+          Each format must be separated by a space or on a new line.
+
+          Example: "zip tar.gz"
+        required: true
+        type: string
+      upload-assets:
+        description: >
+          If true, provenance is uploaded to a GitHub release for new tags.
+          The only value supported is 'true'. The workflow will fail if set to 'false'.
+        required: false
+        type: boolean
+        default: true
```

Comment:
I'm thinking in terms of backward compatibility, eg if we ever want to support non-release archive generation. This would require folks being able to set whether they want to upload  or not, for example. By default it could be that it does not upload, like in other workflows we have. Let me think a bit more, I'm less convinced than I was yesterday :)